### PR TITLE
Faster compiles

### PR DESCRIPTION
**Pull request by robrix**
Speeds up compilation rather dramatically by careful annotation and denormalization.

### CODE REVIEW

File: SwiftCheck/Test.swift [hunk 1]
```diff
@@ -488,8 +488,8 @@ internal func printDistributionGraph(st : State) {
 		return (n < 10 ? " " : "") + "\(n)" + "% "
 	}
 
-	let gAllLabels = st.collected.map({ s in
-		return Array(s).filter({ t in st.labels[t] == .Some(0) }).reduce("", combine: { (l, r) in l + ", " + r })
+	let gAllLabels = st.collected.map({ (s: Set<String>) in
```
Comment (CodaFi) [inline SwiftCheck/Test.swift]: `s : Set<String>`

File: SwiftCheck/Test.swift [hunk 2]
```diff
@@ -488,8 +488,8 @@ internal func printDistributionGraph(st : State) {
...
+		return Array(s).filter({ t in st.labels[t] == .Some(0) }).reduce("", combine: { (l: String, r: String) in l + ", " + r })
```
Comment (CodaFi) [inline SwiftCheck/Test.swift]: Same here

File: SwiftCheck/Test.swift (hunk 1 above)
Comment (robrix) [inline SwiftCheck/Test.swift]: Whoops, my bad—should’ve caught that :grimacing:

Comment (CodaFi) [inline SwiftCheck/Test.swift]: It's no problem.  Our formatting rules are a bit weird, but this one comes from me being paranoid and using too many languages where `<ident>: <type>` isn't allowed.

**Comment by robrix** [comment]
Fixed up the whitespace :+1:

**Comment by robrix** [comment]
Incidentally, this was done with the aid of `-Xfrontend -debug-time-function-bodies` in `OTHER_SWIFT_FLAGS`.

**Comment by CodaFi** [comment]
I don't care how it's been done, you're a god damned magician.

So :boat:

**Comment by robrix** [comment]
:bow:

**Comment by robrix** [comment]
I’m really curious why Property.swift is so slow on Travis. Locally, the slowest single function is < 70ms. For comparison, the fastest one I fixed above was > 2.5s! Maybe an Xcode 6.4b2 vs. Xcode 6.whatever difference?

**Comment by CodaFi** [comment]
Paging @jckarter?

**Comment by robrix** [comment]
A couple of closures on the order of 100ms locally in Xcode 6.3.2. Who knows!

**Comment by robrix** [comment]
Reported to Jo Pamer: https://twitter.com/rob_rix/status/604819847155994625

**Comment by CodaFi** [comment]
Either way... This is definitely a good thing, it going faster in the future.  Good job Swift people!

**Comment by robrix** [comment]
Agreed :clap:

**Pull request merged**